lib/slots.ts

```diff
@@ -1,135 +1,140 @@
 import dayjs, { Dayjs } from "dayjs";
 import utc from "dayjs/plugin/utc";
 import timezone from "dayjs/plugin/timezone";
+import isBetween from "dayjs/plugin/isBetween";
+import { DEFAULT_WORKING_HOURS, WorkingHours } from "@lib/availability";
+
+dayjs.extend(isBetween);
 dayjs.extend(utc);
 dayjs.extend(timezone);
 
-type WorkingHour = {
-  days: number[];
-  startTime: number;
-  endTime: number;
-};
-
-type GetSlots = {
-  inviteeDate: Dayjs;
-  frequency: number;
-  workingHours: WorkingHour[];
-  minimumBookingNotice?: number;
-  organizerTimeZone: string;
-};
-
-type Boundary = {
-  lowerBound: number;
-  upperBound: number;
-};
+type Boundary = [Dayjs, Dayjs];
 
-const freqApply = (cb, value: number, frequency: number): number => cb(value / frequency) * frequency;
+const freqApply = (cb, value: number, frequency: number): number =>
+  value !== 0 ? cb(value / frequency) * frequency : 0;
```

Comment:
I'd rather go 
`value === 0 ? 0 : cb(value / frequency) * frequency` because it appears more linear this way, but that's a subjective thing in the end.